### ISSUE TITLE
Issues with AboutSplatting

### ISSUE THREAD

**Issue by arefg** [vexx32/PSKoans#193]
I'm using this example to demonstrate the problem. For some reason that I couldn't find these two lines are not equal :

                   $x1 = Get-ChildItem C:\Test
                   $x2 = Get-ChildItem C:\Test

running $x1 | Should -Be $x2 will return this error:
            
                  Expected @(Test1-1, Test1-2, Test1-3), but got @(Test1-1, Test1-2, Test1-3).
                  At C:\Program Files\WindowsPowerShell\Modules\Pester\4.8.1\Functions\Assertions\Should.ps1:186 char:9
                  +         throw ( New-ShouldErrorRecord -Message $testResult.FailureMes ...
                  +                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                  + CategoryInfo          : InvalidResult: (System.Collections.Hashtable:Hashtable) [], 
                   Exception
                  + FullyQualifiedErrorId : PesterAssertionFailed

so no matter what format we use for Get-ChildItem parameters (LongLines, Escaping, Hashtable) on lines 28-43, the test will not pass.

I'm not sure if comparing $x1 and $x2 using ($x1 -eq $x2) is right or not, but if it is, it won't return true or false :thinking: and using ($x1 -ne $x2) will print out one of them I think, no true or false again. Maybe this is by design and the reason why the tests don't pass. :thinking: 

One more thing: I think using -Recurse with -Depth doesn't seem to have a meaning, since using -Depth will override the effect of -Recurse. So these will return the same thing:

                   $x1 = Get-ChildItem C:\Test -Depth 2
                   $x2 = Get-ChildItem C:\Test -Recurse -Depth 2

$env:PSKoans:KoanFolder is null:
I'm not sure if this variable populates when running Measure-Karma or not, but running it by itself won't return anything and I think the only reason this command returns the PSKoans folder's content is that I opened up the PSKoans folder in VScode and my current path in terminal is PSKoans folder, so running Get-ChildItem -Path $env:PSKoans:KoanFolder will return the current folder despite -path parameter being null. So I replaced it with $HOME+"\PSKoans" for now.

There is no file in PSKoans root folder (Only folders) so this test will not pass
                    
                  Get-ChildItem @Parameters | Select-Object -First 1 | Should -BeOfType System.IO.FileInfo

It's better to change line 60 to something like this or add the depth parameter in the if statement:

                $Parameters = @{
                       Path = $HOME+"\PSKoans"
                       Depth = 2
                }

I couldn't figure out the last part (Wildcard), I think a little bit more explanation would be helpful.

**Comment by vexx32** [comment]
Yeah, that first one at least is a mistake on my part. Complex objects are rarely considered equal. I should have stripped it down to a single property of the retrieved objects. Good notes on the rest, too!

**Comment by BrandonLundt** [comment]
> $env:PSKoans:KoanFolder is null:
I'm not sure if this variable populates when running Measure-Karma or not, but running it by itself won't return anything and I think the only reason this command returns the PSKoans folder's content is that I opened up the PSKoans folder in VScode and my current path in terminal is PSKoans folder, so running Get-ChildItem -Path $env:PSKoans:KoanFolder will return the current folder despite -path parameter being null. So I replaced it with $HOME+"\PSKoans" for now.

Looks like this has been addressed in #165 

> so no matter what format we use for Get-ChildItem parameters (LongLines, Escaping, Hashtable) on lines 28-43, the test will not pass.

Would comparing one property of these be sufficient? Seems like comparing two would be clear that they are in fact the same. e.g.
```
$Splatted.Count | Should -Be $Escaping.Count
$Splatted.Name | Should -Be $Escaping.Name
            
$Escaping.Count | Should -Be $LongLines.Count
$Escaping.Name | Should -Be $LongLines.Name

$LongLines.Count | Should -Be $Splatted.Count
$LongLines.Name | Should -Be $Splatted.Name
```
